wgengine/router/ifconfig_windows.go

```diff
@@ -622,28 +605,43 @@ func unicastIPNets(ifc *winipcfg.IPAdapterAddresses) []*net.IPNet {
 // doing the minimum number of AddAddresses & DeleteAddress calls.
 // This avoids the full FlushAddresses.
 //
-// Any IPv6 link-local addresses are not deleted.
+// Any IPv6 link-local addresses are not deleted out of caution as some
+// configurations may repeatedly re-add them. Link-local addresses are adjusted
+// to apply SkipAsSource to prevent them from being used, a the addresses will
+// not be routed on the Tailscale interface.
 func syncAddresses(ifc *winipcfg.IPAdapterAddresses, want []*net.IPNet) error {
-	var erracc error
+	var erracc []error
 
 	got := unicastIPNets(ifc)
 	add, del := deltaNets(got, want)
-	del = excludeIPv6LinkLocal(del)
+	ll, del := iter.Split(del, isIPv6LinkLocal)
```

Comment:
Can you do `AppendMatching` and `FilterNotMatching` instead? That way you don't have to allocate two slices

Reply:
You could also do the check in the del loop and not split the slices anyways.